internal/service/ldapconfiguration/resource_ldap_configuration.go

```diff
@@ -238,53 +230,47 @@ func resourceMongoDBAtlasLDAPConfigurationUpdate(ctx context.Context, d *schema.
 		ldap.UserToDNMapping = expandDNMapping(d.Get("user_to_dn_mapping").([]any))
 	}
 
-	ldapReq := &matlas.LDAPConfiguration{
-		LDAP: ldap,
+	params := &admin.UserSecurity{
+		Ldap: ldap,
 	}
-
-	_, _, err := conn.LDAPConfigurations.Save(ctx, d.Id(), ldapReq)
+	_, _, err := connV2.LDAPConfigurationApi.SaveLDAPConfiguration(ctx, d.Id(), params).Execute()
 	if err != nil {
-		return diag.FromErr(fmt.Errorf(errorLDAPConfigurationUpdate, d.Id(), err))
+		return diag.FromErr(fmt.Errorf(errorUpdate, d.Id(), err))
 	}
-
 	return nil
 }
 
-func resourceMongoDBAtlasLDAPConfigurationDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	// Get the client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
-	_, _, err := conn.LDAPConfigurations.Delete(ctx, d.Id())
+func resourceDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
+	_, _, err := connV2.LDAPConfigurationApi.DeleteLDAPConfiguration(ctx, d.Id()).Execute()
 	if err != nil {
-		return diag.FromErr(fmt.Errorf(errorLDAPConfigurationDelete, d.Id(), err))
+		return diag.FromErr(fmt.Errorf(errorDelete, d.Id(), err))
 	}
-
 	return nil
 }
 
-func expandDNMapping(p []any) []*matlas.UserToDNMapping {
-	mappings := make([]*matlas.UserToDNMapping, len(p))
-
+func expandDNMapping(p []any) *[]admin.UserToDNMapping {
+	mappings := make([]admin.UserToDNMapping, len(p))
 	for k, v := range p {
 		mapping := v.(map[string]any)
-		mappings[k] = &matlas.UserToDNMapping{
+		mappings[k] = admin.UserToDNMapping{
 			Match:        mapping["match"].(string),
-			Substitution: mapping["substitution"].(string),
-			LDAPQuery:    mapping["ldap_query"].(string),
+			Substitution: conversion.StringPtr(mapping["substitution"].(string)),
```

Comment:
```suggestion
			Substitution: conversion.Pointer(mapping["substitution"].(string)),
```

Reply:
curious: what is the difference?

Reply:
StringPtr return nil for "" instead of pointer to "",     we may want to revisit when we need StringPtr

Reply:
`StringPtr` returns nil if the string is empty(has zero value). I don't think that in these cases it would make a difference, but conceptually I think it's wrong. As an example, if this was done with an int value(`IntPtr` method), it would return nil if the value was 0, which could be a valid value.
As a side note, I think IntPtr should be removed and will open a PR removing that method and changing from `IntPtr` to `Pointer[int]`

Reply:
changed

Reply:
@oarbusi @marcosuma at the end I need to have StringPtr here and it'll be probably the case for most strings.

In old SDK there were a string with omitempty annotation in json, so it was not sent when "".
now they're a pointer to string, so omitempty doesn't send if it pointer is nil, but if it's pointer to "" then it send "" and this is not an allowed value here. 

So we need StringPtr that return nil for "" so it's not sent.